app/src/org/commcare/heartbeat/HeartbeatWorker.kt

```diff
@@ -0,0 +1,30 @@
+package org.commcare.heartbeat
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import org.commcare.CommCareApplication
+import org.commcare.util.LogTypes
+import org.javarosa.core.services.Logger
+
+/**
+ * @author $|-|!˅@M
+ */
+class HeartbeatWorker(context: Context, workerParams: WorkerParameters):
+        Worker(context, workerParams) {
+    private val requester = CommCareApplication.instance().heartbeatRequester
+    override fun doWork(): Result {
+        try {
+            if (!CommCareApplication.instance().session.heartbeatSucceededForSession()) {
+                requester.makeRequest()
+            }
+        } catch (e: Exception) {
```

Comment:
We should handle different network failures here like we do for other network processes like sync and updates and use Result.retry() for temporary failures.